pkgs/applications/video/mediathekview/default.nix

```diff
@@ -1,49 +1,93 @@
-{ lib, stdenv, fetchurl, makeWrapper, libglvnd, libnotify, jre, zip }:
-
-stdenv.mkDerivation rec {
-  version = "14.0.0";
+{ lib, stdenv, fetchFromGitHub
+, makeShellWrapper, makeDesktopItem, copyDesktopItems, wrapGAppsHook
+, maven, jdk
+, libglvnd, libnotify, libXxf86vm
+}:
+let
+  mavenJdk = maven.override {
+    inherit jdk;
+  };
+in
+mavenJdk.buildMavenPackage rec {
   pname = "mediathekview";
-  src = fetchurl {
-    url = "https://download.mediathekview.de/stabil/MediathekView-${version}-linux.tar.gz";
-    sha256 = "sha256-vr0yqKVRodtXalHEIsm5gdEp9wPU9U5nnYhMk7IiPF4=";
+  version = "14.0.0";
+
+  src = fetchFromGitHub {
+    owner = "mediathekview";
+    repo = "MediathekView";
+    rev = "refs/tags/${version}";
+    hash = "sha256-wVISldd5kixrCVatqI+ZUamh23Kul1QaTk+TDk55NMI=";
   };
 
+  mvnParameters = toString [
+    "-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn"
+    "-Dmaven.test.skip=true"
+    {
+      x86_64-linux = "-Plinux_64bit";
+      aarch64-linux = "-Plinux_arm_64bit";
+    }.${stdenv.hostPlatform.system}
+  ];
 
-  nativeBuildInputs = [ makeWrapper zip ];
+  mvnHash = {
+    x86_64-linux = "sha256-CvzNb9NbZM8ZQS09+Dgn4IUJfl18b/VaVYEDWIU7D+Q=";
+    aarch64-linux = "sha256-N+Q/UONe8GWnC2OjkEtM3oYzdDC3lNCVVuDOLYncPKU=";
+  }.${stdenv.hostPlatform.system};
 
-  installPhase =
-  let
-    libraryPath = lib.strings.makeLibraryPath [ libglvnd libnotify ];
-  in
-  ''
-    runHook preInstall
+  nativeBuildInputs = [
+    wrapGAppsHook
+    copyDesktopItems
+    makeShellWrapper
+  ];
 
-    mkdir -p $out/{bin,lib}
+  preBuild = ''
+    VERSION=${version}
+  '';
 
-    install -m644 MediathekView.jar $out/lib
+  installPhase = ''
+    runHook preInstall
 
-    makeWrapper ${jre}/bin/java $out/bin/mediathek \
-      --add-flags "-jar $out/lib/MediathekView.jar" \
-      --suffix LD_LIBRARY_PATH : "${libraryPath}"
+    mkdir -p $out/{bin,lib}
+    mkdir -p $out/share/icons/hicolor/512x512/apps
 
-    makeWrapper ${jre}/bin/java $out/bin/MediathekView \
-      --add-flags "-jar $out/lib/MediathekView.jar" \
-      --suffix LD_LIBRARY_PATH : "${libraryPath}"
+    install -m644 /build/source/target/MediathekView.jar $out/lib
+    cp ${src}/res/MediathekView.svg $out/share/icons/hicolor/512x512/apps/de.mediathekview.MediathekView.svg
 
-    makeWrapper ${jre}/bin/java $out/bin/MediathekView_ipv4 \
-      --add-flags "-Djava.net.preferIPv4Stack=true -jar $out/lib/MediathekView.jar" \
-      --suffix LD_LIBRARY_PATH : "${libraryPath}"
+    makeShellWrapper ${jdk}/bin/java $out/bin/mediathek \
```

Comment:
Upstream this is MediathekView now. I think we should change it as well.

Reply:
Do you mean to include the mediathek_ipv4? This is not part of the appimage instructions, but mentioned in the troubleshoot guide. My intention was to be close to appimage creation and if somebody runs into IPv6 issues to use overwrite. But I am open to suggestions.